src/NumericalAlgorithms/LinearOperators/PartialDerivatives.cpp

```diff
@@ -293,16 +294,20 @@ GENERATE_INSTANTIATIONS(INSTANTIATION, (1, 2, 3))
 
 #undef INSTANTIATION
 
-#define INSTANTIATION(r, data)                                                \
+#define INSTANTIATE_SCALAR(r, data)                                           \
   template void partial_derivative<Symmetry<>, index_list<>>(                 \
-      gsl::not_null<tnsr::i<DataVector, GET_DIM(data), GET_FRAME(data)>*> du, \
+      const gsl::not_null<                                                    \
+          tnsr::i<DataVector, GET_DIM(data), GET_FRAME(data)>*>               \
+          du,                                                                 \
       const tnsr::i<DataVector, GET_DIM(data), Frame::ElementLogical>&        \
           logical_partial_derivative_of_u,                                    \
       const Mesh<GET_DIM(data)>& mesh,                                        \
       const InverseJacobian<DataVector, GET_DIM(data), Frame::ElementLogical, \
                             GET_FRAME(data)>& inverse_jacobian);              \
   template void partial_derivative(                                           \
-      gsl::not_null<tnsr::i<DataVector, GET_DIM(data), GET_FRAME(data)>*> du, \
+      const gsl::not_null<                                                    \
```

Comment:
strictly speaking you don't need the `const` here, because the const is not part of the function signature (it only makes the variable in the function scope const). Only mentioning this here FYI, I'm not asking you to change this.